src/JSInterop/Microsoft.JSInterop/src/Infrastructure/DotNetDispatcher.cs

```diff
@@ -108,6 +108,20 @@ public static void BeginInvokeDotNet(JSRuntime jsRuntime, DotNetInvocationInfo i
             // Returned a task - we need to continue that task and then report an exception
             // or return the value.
             task.ContinueWith(t => EndInvokeDotNetAfterTask(t, jsRuntime, invocationInfo), TaskScheduler.Current);
+
+        }
+        else if (syncResult?.GetType().GetGenericTypeDefinition() == typeof(ValueTask<>))
+        {
+            //since syncResult is generic we have to get AsTask method for its type
+            var valueTaskAsTaskMethod = syncResult.GetType().GetMethod("AsTask", BindingFlags.Instance | BindingFlags.Public);
+            if (valueTaskAsTaskMethod == null)
+            {
+                throw new MissingMethodException("AsTask method was expected on type ValueTask<>");
+            }
```

Comment:
No need to do this, `Debug.Assert()` at best if you really want it, but if you get here, something is terribly wrong in your app.